docs/data/tree-view/rich-tree-view/selection/SelectionPropagation.tsx

```diff
@@ -0,0 +1,92 @@
+import * as React from 'react';
+import FormControlLabel from '@mui/material/FormControlLabel';
+import Checkbox from '@mui/material/Checkbox';
+import Stack from '@mui/material/Stack';
+import Box from '@mui/material/Box';
+import { RichTreeView } from '@mui/x-tree-view/RichTreeView';
+import {
+  TreeViewBaseItem,
+  TreeViewSelectionPropagation,
+} from '@mui/x-tree-view/models';
+
+const MUI_X_PRODUCTS: TreeViewBaseItem[] = [
+  {
+    id: 'grid',
+    label: 'Data Grid',
+    children: [
+      { id: 'grid-community', label: '@mui/x-data-grid' },
```

Comment:
It'd be great to have a demo with more than two nested level to let the users test propagation better.

Reply:
We don't have a deeper data set yet
I'll take the opportunity to create one :+1:

Reply:
Done :ok_hand: